From 26b363528b56fc980fc4421bc7b09c4855187904 Mon Sep 17 00:00:00 2001
From: Michael Serajnik <m@mser.at>
Date: Sat, 16 Nov 2024 16:31:52 +0100
Subject: [PATCH] Adjust ASCII art.

---
 src/mangosd/Main.cpp | 23 +++++++++++------------
 1 file changed, 11 insertions(+), 12 deletions(-)

diff --git a/src/mangosd/Main.cpp b/src/mangosd/Main.cpp
index cc02792f482..ddb3128b464 100644
--- a/src/mangosd/Main.cpp
+++ b/src/mangosd/Main.cpp
@@ -183,18 +183,17 @@ extern int main(int argc, char **argv)
     sLog.Out(LOG_BASIC, LOG_LVL_MINIMAL, "Core revision: %s [world-daemon]", _FULLVERSION);
     sLog.Out(LOG_BASIC, LOG_LVL_BASIC, "<Ctrl-C> to stop." );
     sLog.Out(LOG_BASIC, LOG_LVL_MINIMAL, "\n\n"
-        "MM   MM         MM   MM  MMMMM   MMMM   MMMMM\n"
-        "MM   MM         MM   MM MMM MMM MM  MM MMM MMM\n"
-        "MMM MMM         MMM  MM MMM MMM MM  MM MMM\n"
-        "MM M MM         MMMM MM MMM     MM  MM  MMM\n"
-        "MM M MM  MMMMM  MM MMMM MMM     MM  MM   MMM\n"
-        "MM M MM M   MMM MM  MMM MMMMMMM MM  MM    MMM\n"
-        "MM   MM     MMM MM   MM MM  MMM MM  MM     MMM\n"
-        "MM   MM MMMMMMM MM   MM MMM MMM MM  MM MMM MMM\n"
-        "MM   MM MM  MMM MM   MM  MMMMMM  MMMM   MMMMM\n"
-        "        MM  MMM http://getmangos.com\n"
-        "        MMMMMM\n\n");
-    sLog.Out(LOG_BASIC, LOG_LVL_MINIMAL, "VMaNGOS : https://github.com/vmangos");
+        "MM     MM MM   MM         MM   MM  MMMMM   MMMMM   MMMMM\n"
+        "MM     MM MM   MM         MM   MM MMM MMM MM   MM MMM MMM\n"
+        " MM   MM  MMM MMM         MMM  MM MMM MMM MM   MM MMM\n"
+        " MM   MM  MM M MM  MMMMM  MMMM MM MMM     MM   MM  MMM\n"
+        "  MM MM   MM M MM M   MMM MM MMMM MMM     MM   MM   MMM\n"
+        "  MM MM   MM M MM     MMM MM  MMM MMMMMMM MM   MM    MMM\n"
+        "   MMM    MM   MM MMMMMMM MM   MM MM  MMM MM   MM     MMM\n"
+        "   MMM    MM   MM MM  MMM MM   MM MMM MMM MM   MM MMM MMM\n"
+        "    M     MM   MM MM  MMM MM   MM  MMMMMM  MMMMM   MMMMM\n"
+        "                  MMMMMM\n"
+        "                          https://github.com/vmangos\n\n");
     sLog.Out(LOG_BASIC, LOG_LVL_MINIMAL, "Using configuration file %s.", cfg_file);
 
     sLog.Out(LOG_BASIC, LOG_LVL_MINIMAL, "Alloc library: " MANGOS_ALLOC_LIB "");
